<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ManipSpaceConfig</class>
 <widget class="QWidget" name="ManipSpaceConfig">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>347</width>
    <height>621</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Planner Space Configuration</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="margin">
    <number>1</number>
   </property>
   <item>
    <widget class="QGroupBox" name="config_space_group">
     <property name="title">
      <string>Configuration Space</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <property name="margin">
       <number>1</number>
      </property>
      <item>
       <widget class="QWidget" name="widget_2" native="true">
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QPushButton" name="load_button">
           <property name="text">
            <string>Load Configs...</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="save_button">
           <property name="text">
            <string>Save Configs...</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Differential Order:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="order_selection">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <item>
         <property name="text">
          <string>0th-Order Space (positions only)</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>1st-Order Space (position + velocity)</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>2nd-Order Space (pos. + vel. + acc.)</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Interpolation Method:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="interp_selection">
        <item>
         <property name="text">
          <string>Linear Interpolation</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Cubic Interpolation</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Quintic Interpolation</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Sustained Velocity Pulses</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Sustained Acceleration Pulses</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QWidget" name="widget" native="true">
        <layout class="QGridLayout" name="gridLayout_2">
         <property name="margin">
          <number>0</number>
         </property>
         <item row="0" column="0">
          <widget class="QCheckBox" name="temporal_space_check">
           <property name="text">
            <string>Temporal Space</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QCheckBox" name="rate_limited_check">
           <property name="text">
            <string>Rate-limited Space</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>Output Diff. Order (e.g. End-effector):</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="output_space_selection">
        <item>
         <property name="text">
          <string>0th-Order Space (positions only)</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>1st-Order Space (position + velocity)</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>2nd-Order Space (pos. + vel. + accel.)</string>
         </property>
        </item>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="travel_interval_group">
     <property name="title">
      <string>Allowed Travel Interval</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <property name="margin">
       <number>1</number>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Maximum:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QDoubleSpinBox" name="max_interval_spinbox">
        <property name="minimum">
         <double>0.100000000000000</double>
        </property>
        <property name="maximum">
         <double>100.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.100000000000000</double>
        </property>
        <property name="value">
         <double>1.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QDoubleSpinBox" name="min_interval_spinbox">
        <property name="minimum">
         <double>0.010000000000000</double>
        </property>
        <property name="maximum">
         <double>20.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.010000000000000</double>
        </property>
        <property name="value">
         <double>0.100000000000000</double>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Minimum:</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>59</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
  <action name="actionValuesChanged">
   <property name="text">
    <string>ValuesChanged</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>interp_selection</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>actionValuesChanged</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>173</x>
     <y>105</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>max_interval_spinbox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>actionValuesChanged</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>258</x>
     <y>236</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>min_interval_spinbox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>actionValuesChanged</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>258</x>
     <y>263</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>order_selection</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>actionValuesChanged</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>173</x>
     <y>58</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>output_space_selection</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>actionValuesChanged</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>173</x>
     <y>177</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rate_limited_check</sender>
   <signal>toggled(bool)</signal>
   <receiver>actionValuesChanged</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>258</x>
     <y>132</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>temporal_space_check</sender>
   <signal>toggled(bool)</signal>
   <receiver>actionValuesChanged</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>88</x>
     <y>132</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
